drivers/bluetooth/hci/spi.c

```diff
@@ -223,14 +224,89 @@ static int bt_spi_get_header(uint8_t op, uint8_t *size)
 	return ret;
 }
 
+#elif DT_HAS_COMPAT_STATUS_OKAY(st_hci_spi_v2)
```

Comment:
I wonder whether would be more readable to have all this routines as callbacks declared in separate files one for each protocol... Not sure it is worth doing it...

Reply:
@avisconti, let me share my point of view about this.
Protocol have all the same structure, that is:
- header phase: used for command and data size definition
- data phase: where data is exchanged

What is more different is that header phase, while data phase is similar.
So, it could be a good compromise to keep everything together.
On the other hand, @JordanYates raised your point in the past as well.

Not a strong opinion from me, but I am little bit more inclined to keep the single file.

Reply:
@msmttchr yes, sure. If changes are not that much, it might be useless to separate into multiple files.